packages/eslint-plugin/rules/__tests__/no-unused-vars-before-return.js

```diff
@@ -0,0 +1,45 @@
+/**
```

Comment:
Note: I named the directory `__tests__` per the behavior of [`requireindex`](https://www.npmjs.com/package/requireindex) to avoid treating "tests" as an export, as it's documented to ignore any entries prefixed with an underscore, and the default Jest preset configuration allows it:

https://github.com/WordPress/gutenberg/blob/837fda74400f6af9810a4ef0b3bb33d22eb0105a/packages/jest-preset-default/jest-preset.json#L12-L16

Alternatively, could consider:

- Using `requireindex`'s second argument to opt in the specific rules exported by `index.js`
- Avoiding `requireindex` for the rules, and building `module.exports` manually
- Defining tests outside the `rules` directory

Reply:
> The optional second argument allows you to explicitly specify the required files using their basename

This mean, it would have to list all folder names to make it work with `test` folder. I think it's fine to keep it as is. In general, `test` is shorter but `__tests__` is also supported to follow what Jest offers by default.